### PR TITLE
Handling all particles with mass 

### PR DESCRIPTION
**Pull request by aemerick**
Updates to handle all particles with mass equally in PMDeposit, PMUpdate, and acceleration instead of just dark matter particles. This should correctly account for particles which have a constant mass and those with a variable mass attribute.

There is one line I'm not sure about, I'll add a note on this.

### CODE REVIEW

File: src/Enzo/enzo_EnzoMethodPmDeposit.cpp [hunk 1]
```diff
@@ -97,221 +105,271 @@ void EnzoMethodPmDeposit::compute ( Block * block) throw()
...
+        ia_m = particle.attribute_index(it, "mass");
+      }
 
-	for (int ip=0; ip<np; ip++) {
+      // Scale mass by volume if particle value is mass instead of density
+      // Required for Cosmology ("mass" is mass)
+      // Not for Collapse ("mass" is density)
+      if (cosmology) {
+        dens *= std::pow(2.0,rank*level);
+      }
```
Comment (aemerick) [inline src/Enzo/enzo_EnzoMethodPmDeposit.cpp:173]: This is the bit I'm not sure about. I had thought particle masses were stored as densities in all cases (at least this is the case for Enzo).... So is this really a cosmology vs. non-cosmology thing or just a quirk of the collapse test?

**Comment by aemerick** [comment]
It may be worth waiting on this until solver-dd gets pulled in since it will generate merge conflicts and I have a version that should be conflict-free with solver-dd... I'd rather make more work for me in dealing with the conflicts than James.

**Comment by forrestglines** [comment]
@aemerick Master still needs to be merged into this branch. The merge conflicts in `src/Enzo/enzo_EnzoMethodPmDeposit.cpp` look non-trivial. Do you still have a version that is more conflict free?

**Comment by gregbryan** [comment]
This looks good to me except for the conflicts.  There is the open question of how the mass/density of particles are dealt with (see Andrew's question above) -- I had a quick look but couldn't be sure; I think we need input from @jobordner on that one.

File: src/Enzo/enzo_EnzoMethodPmDeposit.cpp (hunk 1 above)
Comment (bwoshea) [reply src/Enzo/enzo_EnzoMethodPmDeposit.cpp:173]: @jobordner can you comment on this, please?

**Comment by bwoshea** [comment]
@aemerick would you mind fixing the merge conflicts on this one, please?

**Comment by bwoshea** [comment]
Note: as per my discussion with @aemerick , I believe we are going to scrap this PR because its content is included in PR #49 .  However, since that is still pending we are going to defer merging or declining this PR for the time being.  @aemerick can you confirm?

File: src/Enzo/enzo_EnzoComputeAcceleration.cpp [hunk 2]
```diff
@@ -207,27 +207,35 @@ void EnzoComputeAcceleration::compute_(Block * block)
...
+    std::string particle_type = particle_groups->item("has_mass",ipt);
+    const int it = particle.type_index(particle_type);
+
+    if (particle.num_particles(it) > 0) {
+
+      double dt_shift = 0.5*block->dt() / cosmo_a;
+      //  double dt_shift = 0.0;
+      if (rank_ >= 1) {
+        EnzoComputeCicInterp interp_x
+        	("acceleration_x", particle_type, "ax",dt_shift);
```
Comment (bday336) [inline src/Enzo/enzo_EnzoComputeAcceleration.cpp:226]: Extra space compared to rank 2 and 3

File: src/Enzo/enzo_EnzoMethodPmUpdate.cpp [hunk 3]
```diff
@@ -308,57 +318,76 @@ double EnzoMethodPmUpdate::timestep ( Block * block ) const throw()
...
+        } // rank 1
+
+        if (rank >= 2) {
+      	  enzo_float * vy = (enzo_float *)
+      	    particle.attribute_array (it, ia_vy, ib);
+      	  enzo_float * ay = (enzo_float *)
+      	    particle.attribute_array (it, ia_ay, ib);
+      	  for (int ip=0; ip<np; ip++) {
+      	    double v = fabs(vy[ip*dv]);
+        	  double a = fabs(ay[ip*da]);
```
Comment (bday336) [inline src/Enzo/enzo_EnzoMethodPmUpdate.cpp:364]: Not in line with others (style)

File: src/Enzo/enzo_EnzoMethodPmDeposit.cpp [hunk 4]
```diff
@@ -97,221 +105,271 @@ void EnzoMethodPmDeposit::compute ( Block * block) throw()
...
-	  double x0 = 1.0 - (tx - floor(tx));
-	  double x1 = 1.0 - x0;
 
-	  de_p[ix0] += dens*x0;
-	  de_p[ix1] += dens*x1;
+        if (rank == 1) {
 
-	}
+    	    const int ia_x  = particle.attribute_index(it,"x");
+	        const int ia_vx = particle.attribute_index(it,"vx");
```
Comment (bday336) [inline src/Enzo/enzo_EnzoMethodPmDeposit.cpp:195]: Not in line with each other (style)

**Comment by jwise77** [comment]
Included in #49. Closing.

**Pull request closed without merging**